.github/workflows/aws-snap-integration-gpu.yaml

```diff
@@ -0,0 +1,73 @@
+name: GPU integration CI
+
+on:
+  schedule:
+    - cron: '0 0 * * 0'  # This will trigger the workflow at 00:00 every Sunday
+  workflow_dispatch:
```

Comment:
Is the plan here that we only run these on demand, not as part of the regular CI?

Reply:
Yes .... we conclude on this after syncing with @kimwnasptd ... the gpu instance is quite expensive so we don't want to run it on every PR ... I understand your concern tho.

Reply:
After discussing offline, we've compromised a bit and now have GPU tests running on push but not pull_request.  This means when we push to main we'll at least see the CI fails. 

Longer term, we discussed using labels or something else to run GPU tests only when absolutely needed, but we haven't implemented that here